### ISSUE TITLE
MWER - Pytorch

### ISSUE THREAD

**Issue by zw76859420** [TeaPoly/CTC-MWER#3]
Thank you very much for share the code.
Is there a pytorch version of mwer loss?

**Comment by TeaPoly** [comment]
CE Minimum Word Error Rate Training loss (MWER) based on negative sampling strategy (from <Paraformer: Fast and Accurate Parallel Transformer for Non-autoregressive End-to-End Speech Recognition>):
https://gist.github.com/TeaPoly/234429e6c2d74d10fcb4987bc541d528

CTC based MWER based on PyTorch by @Mddct (I have not verified it):
https://github.com/Mddct/losses

**Comment by zw76859420** [comment]
Thank you very much

**Comment by TeaPoly** [comment]
> Thank you very much for share the code.
> Is there a pytorch version of mwer loss?

You can try `K2` based `MWERLoss`. https://github.com/TeaPoly/CTC-OptimizedLoss/blob/main/torch/mwer_loss.py
